src/atoms/RadioButton/RadioButton.native.js

```diff
@@ -0,0 +1,363 @@
+    },
+  },
+  circle: {
+    dimensions(size) {
+      switch (size) {
+        case 'large':
+          return {
+            width: '14px',
+            height: '14px',
+            borderRadius: '7px',
+            borderWidth: '1.4px',
+          };
+        case 'medium':
+          return {
+            width: '12px',
+            height: '12px',
+            borderRadius: '6px',
+            borderWidth: '1.2px',
+          };
+        case 'small':
+          return {
+            width: '10px',
+            height: '10px',
+            borderRadius: '5px',
+            borderWidth: '1px',
+          };
+        default:
+          return {
+            width: '12px',
+            height: '12px',
+            borderRadius: '6px',
+            borderWidth: '1.2px',
+          };
+      }
+    },
+  },
+  dot: {
+    dimensions(size) {
+      switch (size) {
+        case 'large':
+          return {
+            width: '7px',
+            height: '7px',
+            borderRadius: '3.5px',
+          };
+        case 'medium':
+          return {
+            width: '6px',
+            height: '6px',
+            borderRadius: '3px',
+          };
+        case 'small':
+          return {
+            width: '5px',
+            height: '5px',
+            borderRadius: '2.5px',
+          };
+        default:
+          return {
+            width: '6px',
+            height: '6px',
+            borderRadius: '3px',
+          };
+      }
+    },
+  },
+};
+
+const Dot = styled(View)(
+  (props) =>
+    `
+    border-radius: ${props.borderRadius};
+    background-color: ${props.backgroundColor};
+  `,
+);
+
+const Circle = styled(View)(
+  (props) =>
+    `
+    border-radius: ${props.borderRadius}; 
+    border-color: ${props.color};
+    border-width: ${props.borderWidth}; 
+  `,
+);
+
+const isChecked = (context, currentValue) => {
+  return context && isDefined(context.value) && context.value === currentValue;
+};
+
+const RadioButtonContext = React.createContext(null);
+
+const Radio = ({ value, onChange, defaultValue, children }) => {
+  const [selected, setSelected] = useState(value || defaultValue);
+
+  const onValueChange = useCallback(
+    (newValue) => {
+      if (!value) {
+        setSelected(newValue);
+      }
+      if (isDefined(onChange)) onChange(newValue);
+    },
+    [onChange, value],
+  );
+
+  const contextValue = useMemo(() => ({ value: selected, onChange: onValueChange }), [
+    onValueChange,
+    selected,
+  ]);
+
+  return <RadioButtonContext.Provider value={contextValue}>{children}</RadioButtonContext.Provider>;
+};
+
+const RadioButton = ({
+  size,
+  value,
+  disabled,
+  onClick,
+  title,
+  helpText,
+  errorText,
+  variantColor,
+  testID,
+}) => {
+  const titleTextColor = disabled ? 'shade.950' : 'shade.980';
+  const helpTextColor = disabled ? 'shade.930' : 'shade.950';
+  const [underlayColor, setUnderlayColor] = useState('transparent');
+  const context = useContext(RadioButtonContext);
```

Comment:
throw an error if context is undefined. Basically if someone tries to use `Radio.Option` without `Radio`